### DISCUSSION TITLE
Problem loading data from restarted AIMD trajectories generated using CP2K

### DISCUSSION THREAD

**Discussion by chemrajib** [deepmodeling/dpdata#698] [Q&A] [answered]
Hello All,
I am facing a problem while using dpdata. When I start the simulation from initial structure and perform AIMD run using cp2k, I can load data without any problem. However, when I use any restart point and generate a new sets of trajectories, I am getting the following errors.

> --- You are parsing data using package Cp2kData ---
> Parsing Energies   from ./H2O_restart-1.ener
> Parsing Structures from ./H2O_restart-pos-1.xyz
> Parsing Forces     from ./H2O_restart-frc-1.xyz
> Parsing Stresses   from ./tetramer_restart.log
> Parsing Cells      from ./H2O_restart-1.cell
> Atom names are fake chemical symbols as you set in cp2k input.
> Virial Parsing using cp2kdata as plug in for dpdata
> was not multiplied by volume before cp2kdata v0.6.4
> please check the cp2kdata version and the virial.npy
> 
>  Traceback (most recent call last):
>   File "/home/rajib/DEEPMD/WATER/dpgen_traj/restart_run/dpdata_read.py", line 4, in <module>
>     dp = dpdata.LabeledSystem(cp2kmd_dir, cp2k_output_name=cp2kmd_output_name, fmt="cp2kdata/md")
>   File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 200, in __init__
>     self.from_fmt(
>   File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 237, in from_fmt
>     return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
>   File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 1205, in from_fmt_obj
>     data = fmtobj.from_labeled_system(file_name, **kwargs)
>   File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/dpdata_plugin.py", line 135, in from_labeled_system
>     data['virials'] = cp2kmd.stress_tensor_list*volumes/EV_ANG_m3_TO_GPa
> ValueError: operands could not be broadcast together with shapes (11,3,3) (10,1,1)

I have attached the required files. I will appreciate any help in this regard.
[dpdata_test.zip](https://github.com/user-attachments/files/16723059/dpdata_test.zip)

Thanks!

**Comment by robinzyb** [comment]
please add `restart=True`
```python
dp = dpdata.LabeledSystem(cp2kmd_dir, cp2k_output_name=cp2kmd_output_name, fmt="cp2kdata/md", restart=True)
```

**Reply by robinzyb** [reply]
https://robinzyb.github.io/cp2kdata/docs/dpdata_plugin.html

**Reply by chemrajib** [reply]
Thanks for your response!
Even if I try restart=True flag, its not helping me. I am still getting the same error.

> dp = dpdata.LabeledSystem(cp2kmd_dir, cp2k_output_name=cp2kmd_output_name, fmt="cp2kdata/md", restart=True)
--- You are parsing data using package Cp2kData ---
Parsing Energies   from ./H2O_restart-1.ener
Parsing Structures from ./H2O_restart-pos-1.xyz
Parsing Forces     from ./H2O_restart-frc-1.xyz
Parsing Stresses   from ./tetramer_restart.log
Parsing Cells      from ./H2O_restart-1.cell
Atom names are fake chemical symbols as you set in cp2k input.
Virial Parsing using cp2kdata as plug in for dpdata
was not multiplied by volume before cp2kdata v0.6.4
please check the cp2kdata version and the virial.npy

>Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 200, in __init__
    self.from_fmt(
  File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 237, in from_fmt
    return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
  File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/dpdata/system.py", line 1205, in from_fmt_obj
    data = fmtobj.from_labeled_system(file_name, **kwargs)
  File "/apps/codes/miniforge3/envs/deepmd/lib/python3.10/site-packages/cp2kdata/dpdata_plugin.py", line 135, in from_labeled_system
    data['virials'] = cp2kmd.stress_tensor_list*volumes/EV_ANG_m3_TO_GPa
ValueError: operands could not be broadcast together with shapes (11,3,3) (10,1,1)

**Reply by robinzyb** [reply]
Yes this is a bug, I will fix it later.

**Reply by chemrajib** [reply]
Thanks for your response!

**Reply by robinzyb** [reply]
I have fixed it. Could you provide the restart md calculation with 2 or 3 md steps ? so that I can include your case in my code testing, then I release the new version of cp2kdata. `./dpdata_test/restart_run/`

**Reply by chemrajib** [reply]
Thanks! I have uploaded the files for the restart run here.
[restart_run.zip](https://github.com/user-attachments/files/16737512/restart_run.zip)

In case you need the files for the run from the scratch and inputfiles, I attached here.
[dpdata_test.zip](https://github.com/user-attachments/files/16723059/dpdata_test.zip)

**Reply by chemrajib** [reply]
> I have fixed it. Could you provide the restart md calculation with 2 or 3 md steps ? so that I can include your case in my code testing, then I release the new version of cp2kdata. `./dpdata_test/restart_run/`

Please find the files for 3 md steps restart run.
[restart_run.zip](https://github.com/user-attachments/files/16744367/restart_run.zip)

**Reply by robinzyb** [reply, marked as answer]
Sorry for late reply. Github didn't notify me about your reply with email. https://github.com/robinzyb/cp2kdata/releases/tag/v0.7.1
you may install the updated version with `pip install`. The version number is `v.0.7.1`. Please check the version using `pip check cp2kdata`

**Reply by chemrajib** [reply]
> Sorry for late reply. Github didn't notify me about your reply with email. https://github.com/robinzyb/cp2kdata/releases/tag/v0.7.1 you may install the updated version with `pip install`. The version number is `v.0.7.1`. Please check the version using `pip check cp2kdata`

Thanks for your support. I appreciate your help. This is working now.
